src/focus.service.ts

```diff
@@ -262,12 +268,28 @@ export class FocusService {
   // The client bounding rect when we first selected the element, cached
   // so that we can reuse it if the element gets detached.
   private referenceRect: ClientRect;
+  private focusStack: IFocusState[] = [];
+  private focusRoot: HTMLElement = defaultFocusRoot;
 
   constructor(
     private registry: RegistryService,
     private location: Location,
   ) { }
 
+  public trapFocus(newRootElem: HTMLElement) {
+    this.focusStack.push({
+      root: this.focusRoot,
+      focusedElem: <HTMLElement>document.activeElement,
+    });
+    this.focusRoot = newRootElem;
+  }
+
+  public releaseFocus() {
+    const lastFocusState = this.focusStack.pop();
```

Comment:
yeah thats a good point. I think we should just warn and reset focus instead of throwing error.
Users may land on a page with exiting focus traps.
That said, we should also have a method to clear all traps